packages/camera/camera_android_camerax/lib/src/image_capture.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart' show BinaryMessenger;
+
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+import 'use_case.dart';
+
+/// Use case for picture taking.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/ImageCapture.
+class ImageCapture extends UseCase {
+  /// Creates a [ImageCapture].
+  ImageCapture(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetFlashMode,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageCaptureHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    _api.createFromInstance(this, targetFlashMode, targetResolution);
+  }
+
+  /// Constructs a [ImageCapture] that is not automatically attached to a native object.
+  ImageCapture.detached(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetFlashMode,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageCaptureHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+  }
+
+  late final ImageCaptureHostApiImpl _api;
+
+  /// Flash mode used to take a picture.
+  final int? targetFlashMode;
+
+  /// Target resolution of the image output from taking a picture.
+  final ResolutionInfo? targetResolution;
+
+  /// Constant for automatic flash mode.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_AUTO().
+  static const int flashModeAuto = 0;
+
+  /// Constant for on flash mode.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_ON().
+  static const int flashModeOn = 1;
+
+  /// Constant for no flash mode.
+  ///
+  /// See https://developer.android.com/reference/androidx/camera/core/ImageCapture#FLASH_MODE_OFF().
+  static const int flashModeOff = 2;
+
+  /// Sets the flash mode to use for image capture.
+  void setFlashMode(int newFlashMode) {
```

Comment:
This should return a `Future<void>` and the method itself can just return the call to `_api. setFlashModeFromInstance`.